Docs SEO Updates

keeganhrobinson opened the pull request:
@nguyer this pull request should address the items I've listed in issue #1362. It removes the output from the jekyll-seo-tag plugin and instead uses only the theme to add the page title and various meta tags. It includes custom page titles and meta descriptions for all of the docs pages so they'll be better crawled and displayed by search engines. And last, it hooks up Google Tag Manager so we'll be able to have Google Analytics for the site and see what kind of traffic it's getting. Let me know if you see anything you'd like me to adjust.

keeganhrobinson committed:
Updated page titles and meta descriptions

Signed-off-by: Keegan Robinson <keeganhr@icloud.com>

keeganhrobinson committed:
Add Google Tag Manager

Signed-off-by: Keegan Robinson <keeganhr@icloud.com>

keeganhrobinson committed:
Accounting for autogenerated docs pages

Signed-off-by: Keegan Robinson <keeganhr@icloud.com>

nguyer commented on docs/_includes/head.html:
```diff
@@ -2,28 +2,32 @@
-      <meta name="Description" content="{{ page.description }}">
-    {% endif %}
-  {% endunless %}
+  {% if page.seo_title %}
+    <title>{% t page.seo_title %}</title>
+  {% elsif page.title %}
+    <title>{% t site.name %}: {{ page.title }}</title>
+  {% else %}
+    <title>{% t site.title %}</title>
+  {% endif %}
+  
+  {% if page.seo_description %}
+    <meta name="description" content="{% t page.seo_description %}">
+  {% endif %}
```
Previously there was a distinction between the default page layout, and the translated layout (`i18n_page`), where only the latter used the translation plugin to fill in strings. This meant that if you weren't working with page that was translated, you didn't put any strings in the translations file. This seems to be a change in that behavior in the SEO titles and descriptions always need to be in the translations file.

So one question is, should the SEO information change with the language? If so, that's fine. I'm just not sure.

It also seems a bit strange that I need both the `title` and the `seo_title` now (in different files for a `default` layout) and the `title` is only used in the nav menu. Can we combine these to be the same field?

keeganhrobinson replied:
So I guess I'd say in a dream world you'd want all the content to be translatable. So it's good that the way it is here supports that.

I'm definitely down to continue having `layout: default` available for pages that don't have translations and making it so that it does not require making edits to the translation files. I just tested and that works now but it does break if you try something like `seo_title: "Sample Page Title"` or `seo_description: "Sample page description."`. Maybe I'm missing something obvious but I couldn't find a way to show the non-translated string if it's there in the template. Any chance you have any ideas? I'll keep digging but figured I'd ask.

On the `title` and `seo_title` pieces I do think they should be separate. For the most part, the SEO titles are longer or more descriptive than the titles shown in the navigation. And the way I set it up it will pull in the `title` if there isn't a specific `seo_title` created.

nguyer commented on docs/_i18n/en.yml:
```diff
@@ -58,3 +64,230 @@ pages:
   zksync_testnet: zkSync Testnet
   rotate_dx_certs: Rotate Data Exchange Certificates
+seo:
+  home_title: "Hyperledger FireFly: Decentralized Transaction Processing"
```
Is this subtext used elsewhere to describe FireFly? I'm not sure what the "right" string is here yet, but I've never heard FireFly described as a decentralized transaction processor.

keeganhrobinson replied:
Naw, we can definitely make it better and more in line with what we've said in the past. I'll get this updated when I get the update content.

nguyer commented on docs/_i18n/en.yml:
```diff
@@ -58,3 +64,230 @@ pages:
+seo:
+  home_title: "Hyperledger FireFly: Decentralized Transaction Processing"
+  home_description: "Explore Hyperledger Firefly documentation and features. Learn about distributed ledger technology, smart contracts, and secure data exchange."
+  architecture_title: "Hyperledger FireFly Architecture: Overview and Overview"
```
Maybe just one "Overview"?

keeganhrobinson replied:
Yeah, definitely. Marc's working on combing through these.

nguyer commented on docs/_i18n/en.yml:
```diff
@@ -58,3 +64,230 @@ pages:
+  home_title: "Hyperledger FireFly: Decentralized Transaction Processing"
+  home_description: "Explore Hyperledger Firefly documentation and features. Learn about distributed ledger technology, smart contracts, and secure data exchange."
+  architecture_title: "Hyperledger FireFly Architecture: Overview and Overview"
+  architecture_description: "Hyperledger Firefly Architecture: Discover the decentralized architecture of Firefly for secure and transparent transaction processing."
```
```suggestion
  architecture_description: "Hyperledger FireFly Architecture: Discover the decentralized architecture of Firefly for secure and transparent transaction processing."
```

nguyer commented on docs/_i18n/en.yml:
```diff
@@ -58,3 +64,230 @@ pages:
+  architecture_title: "Hyperledger FireFly Architecture: Overview and Overview"
+  architecture_description: "Hyperledger Firefly Architecture: Discover the decentralized architecture of Firefly for secure and transparent transaction processing."
+  architecture_blockchain_connector_framework_title: "Hyperledger FireFly Architecture: Blockchain Connector Framework"
+  architecture_blockchain_connector_framework_description: "Explore the Blockchain Connector Framework in Hyperledger's FireFly architecture. Learn about the design and functionalities of the framework, which enables seamless integration and interoperability with various blockchain networks and protocols within the FireFly blockchain platform."
```
```suggestion
  architecture_blockchain_connector_framework_description: "Explore the Blockchain Connector Framework in Hyperledger FireFly's architecture. Learn about the design and functionalities of the framework, which enables seamless integration and interoperability with various blockchain networks and protocols within the FireFly blockchain platform."
```

nguyer commented on docs/_config.yml:
```diff
@@ -13,3 +13,4 @@ color_scheme: firefly
 plugins:
   - jekyll-multiple-languages-plugin
 languages: ["en"]
+google_tag_manager: GTM-N58535C
```
We might need to swap this out for a different key. I'm checking on this to see if Hyperledger has one we can use.

keeganhrobinson replied:
Awesome, thanks!

nguyer reviewed (changes requested):
I have a few questions about the mechanics of where stuff is laid out in the docs now. But my bigger concern is the accuracy of the generated descriptions. I started reviewing each description, and after reading several quickly realized they are auto generated. Many of them are "close" to being correct but use terminology that is potentially misleading about what FireFly is or does.

keeganhrobinson commented:
@nguyer I just chatted with Marc and he's going to comb through the titles and descriptions next week so I'll update this to include those when I get them.

keeganhrobinson committed:
Updating apostrophe placement

Co-authored-by: Nicko Guyer <nguyer@users.noreply.github.com>
Signed-off-by: Keegan Robinson <12145278+keeganhrobinson@users.noreply.github.com>
Signed-off-by: Keegan Robinson <keeganhr@icloud.com>

keeganhrobinson committed:
Updating FireFly capitalization

Co-authored-by: Nicko Guyer <nguyer@users.noreply.github.com>
Signed-off-by: Keegan Robinson <12145278+keeganhrobinson@users.noreply.github.com>
Signed-off-by: Keegan Robinson <keeganhr@icloud.com>

keeganhrobinson committed:
Improving SEO support for default layouts

Signed-off-by: Keegan Robinson <keeganhr@icloud.com>

keeganhrobinson committed:
Improving page titles and meta descriptions

Signed-off-by: Keegan Robinson <keeganhr@icloud.com>

nguyer commented:
@keeganhrobinson Is this ready for re-review?

keeganhrobinson commented:
Hey @nguyer! If we can, it'd be nice to bundle in Google Tag Manager with this too. So then we can see what effects—if any—these updates make. If you think that's going to be a while longer though let me take one more pass at this today to make sure everything's good with the latest copy on the site.

nguyer commented:
@SeanBohan Do you know where we're up to with the GTM updates? I think we still need the tag ID to put on the docs site so it can be tracked.

keeganhrobinson committed:
Adding SEO tag support for default layout

keeganhrobinson committed:
Adding open graph meta tags to head

keeganhrobinson committed:
Updating Google Tag Manager to Hyperledger tag

keeganhrobinson commented:
@nguyer I just updated the code so that both the `default` and `i18n_page` layouts work with the SEO meta tags. I also updated the Google Tag Manager to the one you sent along. The last thing for this should just be making sure we get access to that tag manager account so we can use it.

Let me know what you think.

nguyer commented:
Closing this for now as it has been open for a long time with no activity. If we still need it, please feel free to re-open and update the PR. Thanks.
